src/pybind/mgr/rook/module.py

```diff
@@ -248,19 +248,16 @@ def get_hosts(self):
         return [orchestrator.InventoryNode(n, inventory.Devices([])) for n in self.rook_cluster.get_node_names()]
 
     @deferred_read
-    def describe_service(self, service_type=None, service_id=None, node_name=None, refresh=False):
+    def list_daemons(self, daemon_type=None, daemon_id=None, node_name=None, refresh=False):
 
-        if service_type not in ("mds", "osd", "mgr", "mon", "nfs", None):
-            raise orchestrator.OrchestratorValidationError(service_type + " unsupported")
-
-        pods = self.rook_cluster.describe_pods(service_type, service_id, node_name)
+        pods = self.rook_cluster.describe_pods(daemon_type, daemon_id, node_name)
```

Comment:
I will try to take advantage of the current labels in pods and , maybe include new ones, to make easy to get this information.
Using:
https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#list_namespaced_pod
will also simplify things in this method and make it more robust ... 
Let me a little time to check the labels we have now for each service and to propose an alternative implementation based in k8s features to simplify this method

Reply:
Nothing about to get the list of needed pods directly from k8s... forget it. I didn't realize that the source used in describe_pods  is a  KubernetesResource object  that is watching all the time the pods. (I like more to get the things we need in the moment that we want them instead of being watching them all the time, but is just a personal preference)